src/main/java/com/multithreading/TextProcessor.java

```diff
@@ -1,38 +1,67 @@
 package com.multithreading;
 
+import java.util.ArrayList;
+import java.util.List;
+
 class Capitalize implements Runnable {
-    private final char[] chars;
-    private final int index;
+    private volatile boolean isRunning = true;
+    private char character;
```

Comment:
you might want some enum instead or another volatile bool for isCompleted. otherwise your char might not have been processed the the thread is interrupted.